zebra/table_manager.c

```diff
@@ -279,52 +264,17 @@ void table_manager_disable(struct zebra_vrf *zvrf)
 	zvrf->tbl_mgr = NULL;
 }
 
-int table_manager_range(struct vty *vty, bool add, struct zebra_vrf *zvrf,
-			const char *start_table_str, const char *end_table_str)
+void table_manager_range(bool add, struct zebra_vrf *zvrf, uint32_t start,
+			 uint32_t end)
 {
-	uint32_t start;
-	uint32_t end;
-
 	if (add) {
-		if (!start_table_str || !end_table_str) {
-			vty_out(vty, "%% Labels not specified\n");
-			return CMD_WARNING_CONFIG_FAILED;
-		}
-		start = atoi(start_table_str);
-		end = atoi(end_table_str);
-		if (end < start) {
-			vty_out(vty, "%% End table is less than Start table\n");
-			return CMD_WARNING_CONFIG_FAILED;
-		}
+		if (zvrf->tbl_mgr &&
+		    ((zvrf->tbl_mgr->start && zvrf->tbl_mgr->start != start) ||
+		     (zvrf->tbl_mgr->end && zvrf->tbl_mgr->end != end)))
+			zlog_info(
+				"%% New range will be taken into account at restart");
```

Comment:
Does this also require a `wr mem` in order for it to be there on restart? If so it would be nice to mention that here as well.

Reply:
Will do.

Reply:
I moved this check to the CLI handler to show the warning right when the user changes it.